core/lib/zksync_core/src/eth_sender/publish_criterion.rs

```diff
@@ -212,16 +216,20 @@ impl L1BatchPublishCriterion for DataSizeCriterion {
         _storage: &mut StorageProcessor<'_>,
         consecutive_l1_batches: &[L1BatchWithMetadata],
         _last_sealed_l1_batch: L1BatchNumber,
+        l1_batch_commit_data_generator: Arc<dyn L1BatchCommitDataGenerator>,
```

Comment:
Please move `l1_batch_commit_data_generator` to `DataSizeCriterion` so that it doesn't pollute `L1BatchPublishCriterion` interface.

Reply:
Tracked in https://github.com/lambdaclass/zksync-era/issues/133